src/coreComponents/constitutive/solid/PorousSolid.hpp

```diff
@@ -461,16 +187,29 @@ class PorousSolidUpdates : public CoupledSolidUpdates< SOLID_TYPE, BiotPorosity,
 
     updateBiotCoefficient( k );
 
+    // Add the contributions of pressure and temperature to the total stress
     real64 const biotCoefficient = m_porosityUpdate.getBiotCoefficient( k );
+    real64 const thermalExpansionCoefficient = m_solidUpdate.getThermalExpansionCoefficient( k );
+    real64 const bulkModulus = m_solidUpdate.getBulkModulus( k );
+    real64 const thermalExpansionCoefficientTimesBulkModulus = thermalExpansionCoefficient * bulkModulus;
 
-    LvArray::tensorOps::symAddIdentity< 3 >( totalStress, -biotCoefficient * fluidPressure );
+    LvArray::tensorOps::symAddIdentity< 3 >( totalStress, -biotCoefficient * pressure - 3 * thermalExpansionCoefficientTimesBulkModulus * deltaTemperature );
 
+    // Compute derivatives of total stress
     dTotalStress_dPressure[0] = -biotCoefficient;
     dTotalStress_dPressure[1] = -biotCoefficient;
     dTotalStress_dPressure[2] = -biotCoefficient;
     dTotalStress_dPressure[3] = 0;
     dTotalStress_dPressure[4] = 0;
     dTotalStress_dPressure[5] = 0;
+
+    dTotalStress_dTemperature[0] = -3 * thermalExpansionCoefficientTimesBulkModulus;
+    dTotalStress_dTemperature[1] = -3 * thermalExpansionCoefficientTimesBulkModulus;
+    dTotalStress_dTemperature[2] = -3 * thermalExpansionCoefficientTimesBulkModulus;
```

Comment:
See equation 4.21(a) below from Coussy: 

<img width="780" alt="Screen Shot 2022-12-08 at 7 01 20 AM" src="https://user-images.githubusercontent.com/20196406/206480262-0f96bd7e-e421-4a03-ae27-35763e326293.png">

So, if I read well, I am pretty sure we need a minus sign in front of the temperature term in the total stress

Reply:
@sytuannguyen @francoishamon 
Considering the sign convention of stress in GEOSX (negative sign convention for compressive stresses), the minus sign is required for the temperature term when calculating the total stresses.

Reply:
You are right, negative sign convention for compressive stresses so the sign here is negative because temperature increase cause an increase in compressive stress in isochore drained condition.

Reply:
@francoishamon if checking the derivations from Coussy's theory, two different thermal expansion coefficients are used. alpha is used in the stress-strain relation (eq. 4.21a) and alpha_phi is employed in the Biot's porosity model (eq. 4.21c). I guess one refers to the thermal expansion coefficient of the solid phase and the other stands for the thermal expansion coefficient of the fluid/pore-space? But in this PR,  only one `thermalExpansionCoefficient` is defined and used?